File: redistribute-characters-to-make-all-strings-equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        counter = {}
        for word in words:
            for c in word:
                if c not in counter:
                    counter[c] = 0
                counter[c] += 1
        print(counter)
        for k, v in counter.items():
            if v % len(words) != 0:
                return False
        return True

File: redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        unordered_map<char, int> umap;
        for(auto word : words)
            for(char c : word)
                umap[c]++;
        for(auto [c, v] : umap)
            if(v % words.size())
                return false;
        return true;
    }
};
